beacon_node/http_api/src/lib.rs

```diff
@@ -1094,8 +1165,13 @@ pub fn serve<T: BeaconChainTypes>(
                     },
                 };
 
-                Ok(api_types::ExecutionOptimisticResponse {
+                let finalized = chain
+                    .is_finalized_block(&root, block.slot())
+                    .map_err(warp_utils::reject::beacon_chain_error)?;
```

Comment:
We could probably also return the finalization status from `BlockId::root` using similar logic to the logic for states. This may be worthwhile from a performance point of view because the canonical check will likely form a larger fraction of the total runtime for the block-based APIs (blocks are cheap to load, usually a few milliseconds).